homie5/formats/float.yml

```
@@ -1,5 +1,128 @@
 description: validating float formats
 
-# TODO: implement these tests
+tests:
+  - description: proper open-ended minimum format is accepted
+    testtype: propertydescription
+    definition:
+      datatype: float
+      format: "0:"
+    valid: true
 
-tests: []
+  - description: proper open-ended maximum format is accepted
+    testtype: propertydescription
+    definition:
+      datatype: float
+      format: ":100.5"
+    valid: true
+
+  - description: proper closed range format is accepted
+    testtype: propertydescription
+    definition:
+      datatype: float
+      format: "10.1:50.9"
+    valid: true
+
+  - description: proper format with step value is accepted
+    testtype: propertydescription
+    definition:
+      datatype: float
+      format: "0:100:0.5"
+    valid: true
+
+  - description: negative range with step value is accepted
+    testtype: propertydescription
+    definition:
+      datatype: float
+      format: "-100.25:0:0.75"
+    valid: true
+
+  - description: single-point range with step value is accepted
+    testtype: propertydescription
+    definition:
+      datatype: float
+      format: "5.5:5.5:0.1"
+    valid: true
+
+  - description: reverse range is rejected
+    testtype: propertydescription
+    definition:
+      datatype: float
+      format: "50.5:10.2"
+    valid: false
+
+  - description: step value of zero is rejected
+    testtype: propertydescription
+    definition:
+      datatype: float
+      format: "0:100:0"
+    valid: false
+
+  - description: step value larger than range is rejected
+    testtype: propertydescription
+    definition:
+      datatype: float
+      format: "0:10:20"
+    valid: false
+
+  - description: missing range boundaries are rejected
+    testtype: propertydescription
+    definition:
+      datatype: float
+      format: ":"
+    valid: false
```

```suggestion
  - description: missing range boundaries are accepted
    testtype: propertydescription
    definition:
      datatype: float
      format: ":"
    valid: true
```

This one should be accepted, since it is the default format.